### PR TITLE
[ENG-2714] Turn on institutional affiliation by default for registrations

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

On registrations, Institutional affiliation is not on by default in the registration workflow, but is on by default everywhere else on OSF. 

## Changes

- Remove vestigial code that wasn't coping over affiliations correctly 
- Create `excluded_attributes` kwarg for Registration coping function similar to https://github.com/CenterForOpenScience/osf.io/pull/10454
- Clean-up tests and add extra test case
- Clean-up logic in `copy_editable_fields` to not mutate `node` param

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket

https://openscience.atlassian.net/browse/ENG-2714

### CODE REVIEW

File: api/nodes/serializers.py [hunk 1]
```diff
@@ -1618,9 +1611,6 @@ def create(self, validated_data):
         if registration_responses:
             self.update_registration_responses(draft, registration_responses)
 
-        if affiliate_user_institutions and draft.branched_from_type == DraftNode:
```
Comment (Johnetordoff) [inline api/nodes/serializers.py:1621]: This apparently never worked as `branched_from_type` yields a string.

File: osf/models/registrations.py [hunk 2]
```diff
@@ -1261,25 +1261,32 @@ def create_from_node(cls, user, schema, node=None, data=None, provider=None):
...
+            branched_from=branched_from,
             registration_schema=schema,
             registration_metadata=data or {},
             provider=provider,
         )
         draft.save()
-        draft.copy_editable_fields(node, Auth(user), save=True)
+        draft.copy_editable_fields(
+            branched_from,
+            excluded_attributes=excluded_attributes
```
Comment (jwalz) [inline osf/models/registrations.py:1284]: do we need this? I would assume that a DraftNode just has zero Affiliations and therefore wouldn't copy anything?

**Review by jwalz — COMMENTED** [review]
Make sure that this doesn't have conflicts now that #10454 is merged

File: osf/models/registrations.py (hunk 2 above)
Comment (Johnetordoff) [reply osf/models/registrations.py:1284]: You're right we don't need it here.

**Review by jwalz — APPROVED** [review]

**Pull request merged**